### PR TITLE
kv, store: simpilify lazy condition pair

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
It is not necessary to store key and value in condition pair. It may hurt the performance in bulking insert.

### What is changed and how it works?
Change condition pair to an error.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
master:
屏幕快照 2019-08-21 下午7 59 28

this PR:
屏幕快照 2019-08-21 下午8 10 40

the latest commit:
屏幕快照 2019-08-23 下午1 03 57

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by ngaut** [comment]
Good catch.

**Comment by jackysp** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -536,14 +535,14 @@ func (c *twoPhaseCommitter) prewriteSingleBatch(bo *Backoffer, batch batchKeys)
 			// Check already exists error
 			if alreadyExist := keyErr.GetAlreadyExist(); alreadyExist != nil {
 				key := alreadyExist.GetKey()
-				conditionPair := c.txn.us.LookupConditionPair(key)
-				if conditionPair == nil {
-					return errors.Errorf("conn%d, conditionPair for key:%s should not be nil", c.connID, key)
+				existErr := c.txn.us.GetKeyExistErr(key)
+				if existErr == nil {
+					return errors.Errorf("conn%d, existErr for key:%s should not be nil", c.connID, key)
```
Comment (ngaut) [inline store/tikv/2pc.go]: s/conn/conn /

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -536,14 +535,14 @@ func (c *twoPhaseCommitter) prewriteSingleBatch(bo *Backoffer, batch batchKeys)
...
 			if alreadyExist := keyErr.GetAlreadyExist(); alreadyExist != nil {
 				key := alreadyExist.GetKey()
-				conditionPair := c.txn.us.LookupConditionPair(key)
-				if conditionPair == nil {
-					return errors.Errorf("conn%d, conditionPair for key:%s should not be nil", c.connID, key)
+				existErrInfo := c.txn.us.GetKeyExistErrInfo(key)
+				if existErrInfo == nil {
+					return errors.Errorf("conn%d, existErr for key:%s should not be nil", c.connID, key)
 				}
 				logutil.BgLogger().Debug("key already exists",
```
Comment (ngaut) [inline store/tikv/2pc.go]: The debug log message is not need any more. Could you remove it.

**Comment by jackysp** [comment]
> Do we still need to store idx name and value?

We can decode the key and get the index name and value, but I tried it and there are some circular dependencies. Maybe we can wait until the next PR to solve it. @ngaut

**Pull request merged**